### PR TITLE
Add support for frontend v2 for nuxt

### PR DESCRIPTION
**Pull request by vladiulianbogdan**
# Pull Request Template

## Type of change

-   [ ] 🍕 New feature
-   [x] 🐛 Bug Fix
-   [ ] 🔥 Breaking change
-   [ ] 🧑‍💻 Improvement
-   [ ] 📝 Documentation Update

## Description

Add CDN support for Nuxt.

## Checklist

-   [x] My code follows the contributor guidelines of this project;
-   [ ] I have updated the documentation;
-   [ ] I have added tests;
-   [x] New and existing unit tests pass locally with my changes;

### CODE REVIEW

File: src/utils/file.ts [hunk 1]
```diff
@@ -105,6 +105,7 @@ export async function getAllFilesFromCurrentPath(
 export async function zipDirectory(
     sourceDir: string,
     outPath: string,
+    includeHiddenFiles: boolean = false,
```
Comment (andreia-oca) [inline src/utils/file.ts:108]: I think this should be `true` by default.

So before this change `dot: true`, hence all of the previous method calls assume that hidden files will be included by default, right?

File: src/commands/deploy/utils.ts [hunk 2]
```diff
@@ -716,3 +723,31 @@ export async function uploadEnvVarsFromFile(
...
+            "./node_modules",
+            "node_modules",
+            "node_modules/**",
+            "**/node_modules/**",
+        ],
+    );
+
+    await promiseZip;
+    const presignedUrlForProjectCode = await getPresignedURLForProjectCodePush(region, name, stage);
+    return uploadContentToS3(
```
Comment (andreia-oca) [inline src/commands/deploy/utils.ts:749]: I think we should clean `tmpFolderProject` here with something like:
```
await deleteFolder(path.dirname(tmpFolderProject));
```

File: src/commands/deploy/utils.ts [hunk 3]
```diff
@@ -716,3 +723,31 @@ export async function uploadEnvVarsFromFile(
...
+        [
+            "**/node_modules/*",
+            "./node_modules/*",
+            "node_modules/*",
+            "**/node_modules",
+            "./node_modules",
+            "node_modules",
+            "node_modules/**",
+            "**/node_modules/**",
+        ],
```
Comment (andreia-oca) [inline src/commands/deploy/utils.ts:744]: It seems to me that the exclusion list is smaller than it was before refactoring.
I believe we are adding the `.git` directory and any other temporary directories used by vercel, opennnext in the zip.

Is this something we want?

I think we should at least keep ignoring the `.git` related stuff.

File: src/commands/deploy/genezio.ts [hunk 4]
```diff
@@ -279,83 +278,7 @@ export async function genezioDeploy(options: GenezioDeployOptions) {
...
-    const presignedUrlForProjectCode = await getPresignedURLForProjectCodePush(
-        configuration.region,
-        configuration.name,
-        options.stage,
-    );
-    await uploadContentToS3(
-        presignedUrlForProjectCode,
-        path.join(tmpFolderProject, "projectCode.zip"),
-    );
+    await uploadUserCode(configuration.name, configuration.region, options.stage);
```
Comment (costinsin) [inline src/commands/deploy/genezio.ts:281]: Thanks for doing this refactoring! It was DEFINITELY needed.

File: src/commands/deploy/utils.ts (hunk 2 above)
Comment (costinsin) [reply src/commands/deploy/utils.ts:749]: All temporary folders are already deleted at process exit:

https://github.com/Genez-io/genezio/blob/a5e3ea5878b5a41563738926dc30ab8e350ed837/src/index.ts#L40

File: src/commands/deploy/nuxt/deploy.ts [hunk 5]
```diff
@@ -72,16 +97,121 @@ async function deployFunctions(config: YamlProjectConfiguration, stage?: string)
...
+                dot: true,
+                cwd: folder,
+            },
+            (err, files) => {
+                if (err) {
+                    reject(err);
+                    return;
+                }
+
+                const paths: CreateFrontendV2Path[] = files.map((file) => ({
```
Comment (costinsin) [inline src/commands/deploy/nuxt/deploy.ts:207]: What if there are more than 250 files? 😥

**Review by costinsin — APPROVED** [review]
LGTM

File: src/commands/deploy/utils.ts (hunk 2 above)
Comment (andreia-oca) [reply src/commands/deploy/utils.ts:749]: My bad, disregard my comment

File: src/utils/file.ts (hunk 1 above)
Comment (vladiulianbogdan) [reply src/utils/file.ts:108]: Doesn't matter. I changed in all places where this is used. The default value is for future usage and I think false is a more safe default value.

File: src/commands/deploy/utils.ts (hunk 3 above)
Comment (vladiulianbogdan) [reply src/commands/deploy/utils.ts:744]: All dot files and folders are ignored. That's why I did the change with `zipDirectory`.

File: src/utils/file.ts (hunk 1 above)
Comment (andreia-oca) [reply src/utils/file.ts:108]: It's okay to let it be false, but can you also update the following references:

https://github.com/Genez-io/genezio/blob/dev/src/commands/bundle.ts#L117
https://github.com/Genez-io/genezio/blob/dev/src/bundlers/dart/dartBundler.ts#L214

**Review by andreia-oca — APPROVED** [review]
LGTM 🦭

File: src/commands/deploy/nuxt/deploy.ts (hunk 5 above)
Comment (vladiulianbogdan) [reply src/commands/deploy/nuxt/deploy.ts:210]: We cry. I'll add a task for this.

**Review by andreia-oca — APPROVED** [review]

**Pull request merged**